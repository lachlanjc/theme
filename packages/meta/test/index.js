import React from 'react'
import { render, cleanup } from '@testing-library/react'
import Meta from '../src'

afterEach(cleanup)

test('Meta renders', () => {
  const { container, getByText } = render(<Meta />)
  expect(getByText('Hackersent :: Learn cybersecurity')).toBeTruthy()
  expect(
    container.querySelector(
      '[property="og:title"][content="Hackersent :: Learn cybersecurity"]'
    )
  ).toBeTruthy()
  expect(
    container.querySelector(
      '[name="twitter:title"][content="Hackersent :: Learn cybersecurity"]'
    )
  ).toBeTruthy()
  expect(container).toMatchSnapshot()
})

test('Meta renders custom title', () => {
  const title = 'Custom Title :: Hackersent'
  const { container, getByText } = render(<Meta title="Custom Title" />)
  expect(getByText(title)).toBeTruthy()
  expect(
    container.querySelector(`[property="og:title"][content="${title}"]`)
  ).toBeTruthy()
  expect(
    container.querySelector(`[name="twitter:title"][content="${title}"]`)
  ).toBeTruthy()
  expect(container).toMatchSnapshot()
})

test('Meta renders image', () => {
  const url = 'https://hackersent.com/static/card.jpg'
  const { container } = render(<Meta image={url} />)
  expect(
    container.querySelector(
      'meta[name="twitter:card"][content="summary_large_image"]'
    )
  ).toBeTruthy()
  expect(container.querySelectorAll(`[content="${url}"]`)).toHaveLength(2)
  expect(container).toMatchSnapshot()
})

test('Meta renders custom color', () => {
  const color = '#0069ff'
  const { container } = render(<Meta color={color} />)
  expect(
    container.querySelector(`meta[name="theme-color"][content="${color}"]`)
  ).toBeTruthy()
  expect(container).toMatchSnapshot()
})
