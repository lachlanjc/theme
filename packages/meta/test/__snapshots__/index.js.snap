// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Meta renders 1`] = `
<div>
  <meta
    content="en_US"
    property="og:locale"
  />
  <meta
    content="website"
    property="og:type"
  />
  <meta
    content="Hackersent"
    property="og:site_name"
  />
  <meta
    content="@hackersent"
    name="twitter:site"
  />
  <title>
    Hackersent :: Learn cybersecurity
  </title>
  <meta
    content="Hackersent :: Learn cybersecurity"
    property="og:title"
  />
  <meta
    content="Hackersent :: Learn cybersecurity"
    name="twitter:title"
  />
  <meta
    content="#ffffff"
    name="theme-color"
  />
  <meta
    content="#ffffff"
    name="msapplication-TileColor"
  />
  <link
    color="#24292e"
    href="https://hackersent.com/static/favicons/safari-pinned-tab.svg"
    rel="mask-icon"
  />
  <link
    href="https://hackersent.com/static/favicons/apple-touch-icon.png"
    rel="apple-touch-icon"
    sizes="180x180"
  />
  <link
    href="https://hackersent.com/static/favicons/favicon-32x32.png"
    rel="icon"
    sizes="32x32"
    type="image/png"
  />
  <link
    href="https://hackersent.com/static/favicons/favicon-16x16.png"
    rel="icon"
    sizes="16x16"
    type="image/png"
  />
  <link
    href="https://hackersent.com/static/site.webmanifest"
    rel="manifest"
  />
</div>
`;

exports[`Meta renders custom color 1`] = `
<div>
  <meta
    content="en_US"
    property="og:locale"
  />
  <meta
    content="website"
    property="og:type"
  />
  <meta
    content="Hackersent"
    property="og:site_name"
  />
  <meta
    content="@hackersent"
    name="twitter:site"
  />
  <title>
    Hackersent :: Learn cybersecurity
  </title>
  <meta
    content="Hackersent :: Learn cybersecurity"
    property="og:title"
  />
  <meta
    content="Hackersent :: Learn cybersecurity"
    name="twitter:title"
  />
  <meta
    content="#0069ff"
    name="theme-color"
  />
  <meta
    content="#0069ff"
    name="msapplication-TileColor"
  />
  <link
    color="#24292e"
    href="https://hackersent.com/static/favicons/safari-pinned-tab.svg"
    rel="mask-icon"
  />
  <link
    href="https://hackersent.com/static/favicons/apple-touch-icon.png"
    rel="apple-touch-icon"
    sizes="180x180"
  />
  <link
    href="https://hackersent.com/static/favicons/favicon-32x32.png"
    rel="icon"
    sizes="32x32"
    type="image/png"
  />
  <link
    href="https://hackersent.com/static/favicons/favicon-16x16.png"
    rel="icon"
    sizes="16x16"
    type="image/png"
  />
  <link
    href="https://hackersent.com/static/site.webmanifest"
    rel="manifest"
  />
</div>
`;

exports[`Meta renders custom title 1`] = `
<div>
  <meta
    content="en_US"
    property="og:locale"
  />
  <meta
    content="website"
    property="og:type"
  />
  <meta
    content="Hackersent"
    property="og:site_name"
  />
  <meta
    content="@hackersent"
    name="twitter:site"
  />
  <title>
    Custom Title :: Hackersent
  </title>
  <meta
    content="Custom Title :: Hackersent"
    property="og:title"
  />
  <meta
    content="Custom Title :: Hackersent"
    name="twitter:title"
  />
  <meta
    content="#ffffff"
    name="theme-color"
  />
  <meta
    content="#ffffff"
    name="msapplication-TileColor"
  />
  <link
    color="#24292e"
    href="https://hackersent.com/static/favicons/safari-pinned-tab.svg"
    rel="mask-icon"
  />
  <link
    href="https://hackersent.com/static/favicons/apple-touch-icon.png"
    rel="apple-touch-icon"
    sizes="180x180"
  />
  <link
    href="https://hackersent.com/static/favicons/favicon-32x32.png"
    rel="icon"
    sizes="32x32"
    type="image/png"
  />
  <link
    href="https://hackersent.com/static/favicons/favicon-16x16.png"
    rel="icon"
    sizes="16x16"
    type="image/png"
  />
  <link
    href="https://hackersent.com/static/site.webmanifest"
    rel="manifest"
  />
</div>
`;

exports[`Meta renders image 1`] = `
<div>
  <meta
    content="en_US"
    property="og:locale"
  />
  <meta
    content="website"
    property="og:type"
  />
  <meta
    content="Hackersent"
    property="og:site_name"
  />
  <meta
    content="@hackersent"
    name="twitter:site"
  />
  <title>
    Hackersent :: Learn cybersecurity
  </title>
  <meta
    content="Hackersent :: Learn cybersecurity"
    property="og:title"
  />
  <meta
    content="Hackersent :: Learn cybersecurity"
    name="twitter:title"
  />
  <meta
    content="https://hackersent.com/static/card.jpg"
    property="og:image"
  />
  <meta
    content="summary_large_image"
    name="twitter:card"
  />
  <meta
    content="https://hackersent.com/static/card.jpg"
    name="twitter:image"
  />
  <meta
    content="#ffffff"
    name="theme-color"
  />
  <meta
    content="#ffffff"
    name="msapplication-TileColor"
  />
  <link
    color="#24292e"
    href="https://hackersent.com/static/favicons/safari-pinned-tab.svg"
    rel="mask-icon"
  />
  <link
    href="https://hackersent.com/static/favicons/apple-touch-icon.png"
    rel="apple-touch-icon"
    sizes="180x180"
  />
  <link
    href="https://hackersent.com/static/favicons/favicon-32x32.png"
    rel="icon"
    sizes="32x32"
    type="image/png"
  />
  <link
    href="https://hackersent.com/static/favicons/favicon-16x16.png"
    rel="icon"
    sizes="16x16"
    type="image/png"
  />
  <link
    href="https://hackersent.com/static/site.webmanifest"
    rel="manifest"
  />
</div>
`;
